### PR TITLE
[Layout] Fix ASDimension creation of type ASDimensionUnitFraction via a NSString

### PR DESCRIPTION
**Pull request by maicki**

### CODE REVIEW

**Comment by maicki** [comment]
@Adlai-Holler Ready for review.

**Comment by Adlai-Holler** [comment]
In a future diff, let's add `ASXCTAssertEqualDimensions` to our custom test headers implementation, so if a test fails we can find out why.

**Comment by Adlai-Holler** [comment]
Feel free to merge after CI passes. I'm going to be OOO for a little bit.

File: AsyncDisplayKit/Layout/ASDimension.mm [hunk 1]
```diff
@@ -38,7 +38,7 @@ ASOVERLOADABLE ASDimension ASDimensionMake(NSString *dimension)
     // Check if percent is at the end and remove it
```
Comment (appleguy) [inline AsyncDisplayKit/Layout/ASDimension.mm]: @maicki above, in the dimension.length == 0 case, should we maybe return ASDimensionAuto and assert?

**Comment by appleguy** [comment]
Thanks for adding the tests, @maicki !  I also really like having NSStringFromASDimension().

The failure is the layout attributes thing...

ASCollectionViewTests
  testSelection, failed: caught "NSInternalInconsistencyException"
  /Users/travis/build/facebook/AsyncDisplayKit/AsyncDisplayKitTests/ASCollectionViewTests.mm:97
  ```
{
  ASDisplayNodeAssertNotNil(node.layoutAttributes, @"Expected layout attributes for node in %@ to be non-nil.", NSStringFromSelector(_cmd));

File: AsyncDisplayKit/Layout/ASDimension.mm (hunk 1 above)
Comment (maicki) [inline AsyncDisplayKit/Layout/ASDimension.mm]: Change is in here: #2351

**Comment by maicki** [comment]
@Adlai-Holler Added `ASXCTAssertEqualDimensions` in #2351

**Pull request merged**